parlai/core/torch_agent.py

```diff
@@ -285,14 +285,16 @@ def batchify(self, obs_batch, sort=False,
 
         if sort:
             ind_sorted = sorted(range(len(x_lens)), key=lambda k: -x_lens[k])
-
             exs = [exs[k] for k in ind_sorted]
             valid_inds = [valid_inds[k] for k in ind_sorted]
             x_text = [x_text[k] for k in ind_sorted]
+            x_lens = [x_lens[k] for k in ind_sorted]
```

Comment:
This is to fix a small bug. The lengths weren't ordered in the same order as the text fields, which caused errors with `pack_padded_sequence`.